Automattic/wp-calypso
# Activity log: Validate API data with schema

Activity item validation in `fromApi` was very weak. This PR vastly improves the item validation by leveraging the data-layer `fromApi` and `makeParser`.

Name and group schema enum validation have been removed from the persistence schema as we're being careful to ensure we have "good" data at the point of entry.

## Testing
1. Visit https://calypso.live/stats/activity?branch=update/activity-log/api-item-schema
1. Do you continue to see activity in the log?
1. Try refreshing. Is activity persisted as expected?
1. ~You will likely see console warnings from the schema validation. This is expected as unrecognized activities are being returned from the API. Are any of them unexpected?~
